sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -273,27 +273,32 @@ private void flush() {
     }
 
     private void exportBatches(ArrayList<ReadableSpan> spanList) {
-      ArrayList<Span> spansProtoList = new ArrayList<>(maxExportBatchSize);
       // TODO: Record a counter for pushed spans.
       for (int i = 0; i < spanList.size(); ) {
+        List<SpanData> spanDataBuffer = new ArrayList<>(maxExportBatchSize);
```

Comment:
Move this to createSpanDataForExport and make it return a List<SpanData>?

Reply:
yes, definitely. @breedx-nr and I had discussed exactly that. Must have missed it in the final implementation.

Reply:
done